src/transformers/utils/generic.py

```diff
@@ -902,3 +908,70 @@ def is_timm_local_checkpoint(pretrained_model_path: str) -> bool:
         return is_timm_config_dict(config_dict)
 
     return False
+
+
+def set_arg_to_config_and_sub_configs(config, key, value):
+    """
+    Set a value to a config and all sub configs.
+    """
+    from ..configuration_utils import PretrainedConfig
+
+    setattr(config, key, value)
+    for attribute_name in dir(config):
+        attribute = getattr(config, attribute_name)
+        if isinstance(attribute, PretrainedConfig):
+            set_arg_to_config_and_sub_configs(attribute, key, value)
+
+
+def del_arg_from_config_and_sub_configs(config, key):
+    """
+    Delete a value from a config and all sub configs.
+    """
+    from ..configuration_utils import PretrainedConfig
+
+    delattr(config, key)
+    for attribute_name in dir(config):
+        attribute = getattr(config, attribute_name)
+        if isinstance(attribute, PretrainedConfig):
+            del_arg_from_config_and_sub_configs(attribute, key)
+
+
+def can_return_tuple(func):
+    """
+    Decorator to wrap model method, to call output.to_tuple() if return_dict=False passed as a kwarg or
+    use_return_dict=False is set in the config.
+
+    Note:
+        output.to_tuple() convert output to tuple skipping all `None` values.
+    """
+
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
```

Comment:
```suggestion
    def wrapper(self, *args, **kwargs) -> Union[tuple, BaseModelOuptut:
```
don't know if typing can be made better for this one?

Reply:
No need to specify return type hint here, typing should be copied from the wrapped method, cause we have a `@warps` decorator